### PR TITLE
IDE-594 Import of .DUD file from KEL fails and gives boost warning

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Fixed IDE-594

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith I'm waiting for Eric to test this. I still have not rebased as well. I would like you to look at my solution to make sure I'm not far off base before going forward.

File: comms/ModFileRepository.cpp [hunk 1]
```diff
@@ -240,10 +249,24 @@ class CModFileRepository : public IRepository, public clib::CLockableUnknown
...
                 attributeLabel = trimmedLine;
+
+                std::vector<std::_tstring> parts;
+                boost::split(parts, attributeLabel, boost::is_any_of(":"), boost::token_compress_on);
+                if (parts.size() > 1) {
+                    attributeLabel = parts[1];
+                    attributeExtension = parts[0];
+                }
+                else {
+                    attributeExtension = _T("ECL");
```
Comment (GordonSmith) [inline comms/ModFileRepository.cpp]: Is there no label in this scenario?

Comment (dehilsterlexis) [inline comms/ModFileRepository.cpp]: In line 258, it is initialized. attributeLabel = trimmedLine;

File: comms/ModFileRepository.cpp [hunk 2]
```diff
@@ -221,17 +221,26 @@ class CModFileRepository : public IRepository, public clib::CLockableUnknown
     unsigned GetAttributes(const TCHAR* module, const IAttributeTypeVector & types, IAttributeVector & attributes, bool noRefresh=false, bool noBroadcast=false) const
     {
         clib::recursive_mutex::scoped_lock proc(m_mutex);
-        typedef std::pair<std::_tstring, std::_tstring> CommentEclPair;
-        typedef std::map<std::_tstring, CommentEclPair> AttrMap;
+        //typedef std::pair<std::_tstring, std::_tstring> CommentEclPair;
```
Comment (GordonSmith) [inline comms/ModFileRepository.cpp]: Remove commented code.

File: comms/ModFileRepository.cpp (hunk 1 above)
Comment (GordonSmith) [inline comms/ModFileRepository.cpp]: ok

**Comment by GordonSmith** [comment]
Waiting on: 
1. Removal of commented code
2. More testing and signoff from @eblood66

**Comment by richardkchapman** [comment]
Needs rebasing for a clean merge

**Comment by dehilsterlexis** [comment]
@richardkchapman please do not merge. I have to fix this PR. It has IDE-373 inside it so it is wrong. Will ping you when I get this resolved.

**Comment by dehilsterlexis** [comment]
@GordonSmith @richardkchapman Eric Blood has confirmed this fixes the problem. It has been rebased and is ready.

**Comment by richardkchapman** [comment]
@GordonSmith Please review

**Pull request merged**